File: convert-json-to-md.py
import os
import json

export_dir = "export"
files = [f for f in os.listdir(export_dir) if f.endswith(".json")]
converted_count = 0

for file in files:
    file_path = os.path.join(export_dir, file)
    with open(file_path, "r", encoding="utf-8") as f:
        try:
            data = json.load(f)
        except Exception as e:
            print(f"Error parsing {file}: {e}")
            continue

    base_name = file.replace(".json", "")
    md_content = f"# {base_name} Data Export\n\n"

    if not isinstance(data, list):
        md_content += f"```json\n{json.dumps(data, indent=2)}\n```\n"
    else:
        md_content += f"**Total records:** {len(data)}\n\n"
        
        if len(data) > 0:
            keys = set()
            for item in data:
                if isinstance(item, dict):
                    keys.update(item.keys())
            
            columns = list(keys)
            
            # Header
            md_content += "| " + " | ".join(columns) + " |\n"
            md_content += "| " + " | ".join(["---"] * len(columns)) + " |\n"
            
            # Rows
            for item in data:
                if not isinstance(item, dict):
                    md_content += f"| {str(item)} |\n"
                    continue
                
                row = []
                for col in columns:
                    val = item.get(col)
                    if val is None:
                        row.append("")
                    elif isinstance(val, (dict, list)):
                        row.append(json.dumps(val).replace("|", "\\|"))
                    else:
                        str_val = str(val).replace("|", "\\|").replace("\n", "<br>")
                        if len(str_val) > 500:
                            str_val = str_val[:500] + "..."
                        row.append(str_val)
                
                md_content += "| " + " | ".join(row) + " |\n"
        else:
            md_content += "*No records found in this file.*\n"

    out_path = os.path.join(export_dir, f"{base_name}.md")
    with open(out_path, "w", encoding="utf-8") as f:
        f.write(md_content)
    
    converted_count += 1

print(f"Successfully converted {converted_count} JSON files to Markdown.")

File: convert-json-to-md.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const exportDir = path.join(__dirname, 'export');
const files = fs.readdirSync(exportDir).filter(f => f.endsWith('.json'));

let convertedCount = 0;

for (const file of files) {
  const filePath = path.join(exportDir, file);
  const data = JSON.parse(fs.readFileSync(filePath, 'utf8'));
  
  const baseName = file.replace('.json', '');
  let md = `# ${baseName} Data Export\n\n`;
  
  if (!Array.isArray(data)) {
    // Some files might be single objects
    md += `\`\`\`json\n${JSON.stringify(data, null, 2)}\n\`\`\`\n`;
  } else {
    md += `**Total records:** ${data.length}\n\n`;
    
    if (data.length > 0) {
      // Find all unique keys across all objects to ensure we don't miss columns
      const keys = new Set();
      data.forEach(item => {
        if (typeof item === 'object' && item !== null) {
          Object.keys(item).forEach(k => keys.add(k));
        }
      });
      const columns = Array.from(keys);
      
      // Markdown Table Header
      md += `| ${columns.join(' | ')} |\n`;
      md += `| ${columns.map(() => '---').join(' | ')} |\n`;
      
      // Markdown Table Rows
      data.forEach(item => {
        if (typeof item !== 'object' || item === null) {
            // handle arrays of primitives if any
            md += `| ${String(item)} |\n`;
            return;
        }

        const row = columns.map(col => {
          let val = item[col];
          if (val === null || val === undefined) return '';
          if (typeof val === 'object') {
            return JSON.stringify(val).replace(/\|/g, '\\|');
          }
          let strVal = String(val);
          // Escape markdown table characters
          strVal = strVal.replace(/\|/g, '\\|').replace(/\n/g, '<br>');
          // Cap very long strings to keep table readable
          if (strVal.length > 500) {
            strVal = strVal.substring(0, 500) + '...';
          }
          return strVal;
        });
        md += `| ${row.join(' | ')} |\n`;
      });
    } else {
      md += `*No records found in this file.*\n`;
    }
  }
  
  fs.writeFileSync(path.join(exportDir, `${baseName}.md`), md);
  convertedCount++;
}

console.log(`Successfully converted ${convertedCount} JSON files to Markdown.`);
